spec/integration/connection_timeout_spec.rb

```diff
@@ -0,0 +1,32 @@
+if ENV['INTEGRATION_TIMEOUT']
```

Comment:
Is this environment flag necessary anymore? Sorry, I wasn't clear earlier but I was thinking this environment flag would get removed after we pulled 1.8.7 support.

Reply:
My bad, indeed that makes sense. Rebased.